rss-reader/Model/SourceManager/Repository/UVSourceRepository.m

```diff
@@ -0,0 +1,55 @@
+//
+//  UVSourceRepository.m
+//  rss-reader
+//
+//  Created by Uladzislau Volchyk on 22.12.20.
+//
+
+#import "UVSourceRepository.h"
+#import "KeyConstants.h"
+#import "NSArray+Util.h"
+
+@interface UVSourceRepository ()
+
+@property (nonatomic, copy) NSString *path;
+
+@end
+
+@implementation UVSourceRepository
+
+- (instancetype)initWithPath:(NSString *)path
+{
+    self = [super init];
+    if (self) {
+        _path = [path copy];
+    }
+    return self;
+}
+
+- (void)dealloc
+{
+    [_path release];
+    [super dealloc];
+}
+
+- (NSArray<NSDictionary *> *)fetchData:(out NSError **)error {
+    NSData *data = [NSData dataWithContentsOfFile:self.path];
+    if (!data) {
+        return @[];
+    }
+    return [NSPropertyListSerialization propertyListWithData:data
+                                                     options:0
+                                                      format:nil
+                                                       error:error];
+}
+
+- (BOOL)updateData:(NSArray<NSDictionary *> *)data error:(out NSError **)error {
+    NSData *plist = [NSPropertyListSerialization dataWithPropertyList:data
+                                                               format:NSPropertyListXMLFormat_v1_0
+                                                              options:0
+                                                                error:error];
+    [plist writeToFile:self.path atomically:YES];
+    return error == nil;
```

Comment:
Error is passed here by reference. You will get false return value in case when there was no errors in `dataWithPropertyList:...` method. True only if nil will passed as a parameter.
You are should check for nil *error here.

Reply:
fixed